boards/arm/nucleo_g0b1re/nucleo_g0b1re.dts

```diff
@@ -127,6 +128,40 @@
 	pinctrl-0 = <&dac1_out1_pa4>;
 };
 
+&shared_irq21 {
+	status = "okay";
+};
+
+&shared_irq22 {
+	status = "okay";
+};
+
+&can1 {
+	pinctrl-0 = <&fdcan1_rx_pa11 &fdcan1_tx_pa12>;
+	shared-irqs = <&shared_irq21 &shared_irq22>;
+	shared-irq-names = "LINE_0", "LINE_1";
+	bus-speed = <125000>;
+	sjw = <1>;
+	sample-point = <875>;
+	bus-speed-data = <1000000>;
+	sjw-data = <1>;
+	sample-point-data = <875>;
+	status = "okay";
+};
+
+&can2 {
+	pinctrl-0 = <&fdcan2_rx_pb5 &fdcan2_tx_pb6>;
+	shared-irqs = <&shared_irq21>, <&shared_irq22>;
+	shared-irq-names = "LINE_0", "LINE_1";
```

Comment:
I wonder why these are defined at board level.
Is there some level of configuration that is left to the user?

Reply:
My initial idea was that irq conflicts arise depending on which peripherals are used in the application.
In cases where only CAN1 is used there is no need to share the interrupt.
In cases where only CAN1 and TIM16 are used irq21 could be shared while irq22 is only used by CAN1 line_1.

However this has the downside that additional configuration by the "user" is needed.
Would you prefer if the driver would simply use shared-irqs all the time, or enable shared-irq support by another means?

Reply:
ok, I get your point.

Actually I was not sure you considered the case of having different IRQ scheme depending on conflicts possibility (looking at the modifications done to `config_can_##inst##_irq` function.

I'd prefer to avoid having users configuring this. Device tree is already complex for new comers, and this would add one more significant step.

I see following options:
- A] Keep shared IRQ used even if there is no conflict in user configuration
    (+) Simple to handle at driver level
    (+) Same code used with or w/o conflicting devices enabled. Which is better in general for analysis if real time issues
    (-) Useless footprint/performance overhead 
    (-) Different real time behavior of potentially conflicting nodes and no conflict nodes
- B] Detect conflicting configurations (using DT macros) and enable shared IRQ only when required
    (+) Performance/Footprint optimization
    (-) Increased complexity at driver level
    (-) Different code used in both cases 
 - C] User configurable B] (same as B], but with Kconfig option that allows to always use shared IRQ, even if no conflict is possible)
   (+) User can choose between Performance/Footprint optimization and similar real time behavior
   (-) Driver complexity ++

Reply:
Thanks for this detailed listing of options.
I have now as a first step moved the lines highlighted above into the soc's dtsi.
There was really no good reason not to have them already defined. Now shared-irqs are used if the referenced shared-irq is enabled, meaning the user would still have to set `status = "okay"` for the relevant irq nodes.
Before I comment on above options, may I ask if
> I'd prefer to avoid having users configuring this. 

also refers to setting `status = "okay"` for the shared-irq nodes?

Reply:
> also refers to setting status = "okay" for the shared-irq nodes?

It depends on the solution chosen in the end, but if we can minimize the possibility of errors for end users, it would be great